lib/connection/result/datetime_format_converter.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+function formatTagsMap() {
+  return [
+    // proper mappings
+    ['YYYY', 'YYYY'],
+    ['YY', 'YY'],
+    ['MM', 'MM'],
+    ['MON', 'MMM'],
+    ['DD', 'DD'],
+    ['DY', 'ddd'],
+    ['HH24', 'HH'],
+    ['HH12', 'hh'],
+    ['HH', 'HH'],
+    ['AM', 'A'],
+    ['PM', 'A'],
+    ['MI', 'mm'],
+    ['SS', 'ss'],
+    ['TZH:TZM', 'Z'],
+    ['TZHTZM', 'ZZ'],
+
+    // special code needed
+    ['TZH', ''],
+    ['TZM', ''],
+    ['FF', '']
+  ];
+}
+
+function convertSnowflakeFormatToMomentFormat(formatSql, scale) {
+  const tags = formatTagsMap();
+
+  // get an upper-case version of the input sql format
+  const formatSqlUpper = formatSql.toUpperCase();
+  let moment;
+
+  // iterate over the format string
+  const length = formatSql.length;
+  let formatMoment = '';
+  for (let pos = 0; pos < length;) {
+    let tag = null;
+    let out = null;
+
+    // at each position, check if there's a tag at that position; if so, use
+    // 'out' as the replacement
+    for (let index = 0; index < tags.length; index++) {
+      if (formatSqlUpper.substr(pos).indexOf(tags[index][0]) === 0) {
+        tag = tags[index][0];
+        out = tags[index][1];
+        break;
+      }
+    }
+
+    // if we didn't find a match, just insert the character after escaping it
+    // (by wrapping it in square brackets)
+    if (out === null) {
+      formatMoment += formatSql[pos];
+      pos++;
+    } else {
+      // we found one of our special tags
+      if (out === '') {
+        if (tag === 'TZH') {
+          // format the moment to get the timezone string and extract the
+          // hours; for example, '-0700' will be converted to '-07'
+          out = moment.format('ZZ').substr(0, 3);
```

Comment:
maybe I missed something but the moment is declared as variable in line 36 but i don't see where it's assigned. Wouldn't it always throw an error here due to being undefined?

Reply:
Thanks @sfc-gh-mkubik This part wasn't used. I removed it.